src/Summoner/CLI.hs

```diff
@@ -0,0 +1,181 @@
+travisP d = flag Idk d
+          $ long "travis"
+         <> short 'c'
+         <> help "Travis CI integration"
+
+appVeyorP :: Decision -> Parser Decision
+appVeyorP d = flag Idk d
+            $ long "app-veyor"
+           <> short 'w'
+           <> help "AppVeyor CI integration"
+
+privateP :: Decision -> Parser Decision
+privateP d = flag Idk d
+           $ long "private"
+          <> short 'p'
+          <> help "Private repository"
+
+scriptP :: Decision -> Parser Decision
+scriptP d = flag Idk d
+          $ long "script"
+         <> short 's'
+         <> help "Build script for convenience"
+
+libraryP :: Decision -> Parser Decision
+libraryP d = flag Idk d
+           $ long "library"
+          <> short 'l'
+          <> help "Library folder"
+
+execP :: Decision -> Parser Decision
+execP d = flag Idk d
+        $ long "exec"
+       <> short 'e'
+       <> help "Executable target"
+
+testP :: Decision -> Parser Decision
+testP d =  flag Idk d
+        $  long "test"
+        <> short 't'
+        <> help "Test target"
+
+benchmarkP :: Decision -> Parser Decision
+benchmarkP d = flag Idk d
+             $ long "benchmark"
+            <> short 'b'
+            <> help "Benchmarks"
+
+onP :: Parser Targets
+onP  = subparser $ mconcat
+       [ metavar "on [OPTIONS]"
+       , command "on" $ info (helper <*> targetsP Yes) (progDesc "Specify options to enable")
+       ]
+
+offP :: Parser Targets
+offP = subparser $ mconcat
+       [ metavar "off [OPTIONS]"
+       , command "off" $ info (helper <*> targetsP Nop) (progDesc "Specify options to disable")
+       ]
+
+optsP :: Parser InitOpts
+optsP = do
+    projectName <- T.pack <$> strArgument (metavar "PROJECT_NAME")
+    on  <- optional onP
+    off <- optional offP
+
+    pure $ InitOpts projectName (fold $ on `mappend` off)
+
+prsr :: ParserInfo InitOpts
+prsr = modifyHeader
+     $ modifyFooter
+     $ info ( helper <*> optsP )
+            $ fullDesc
+           <> progDesc "Create your own haskell project"
+
+-- to put custom header which doesn't cut all spaces
+modifyHeader :: ParserInfo InitOpts -> ParserInfo InitOpts
+modifyHeader initOpts = initOpts {infoHeader = stringChunk $ T.unpack artHeader}
+
+-- to put custom footer which doesn't cut all spaces
+modifyFooter :: ParserInfo InitOpts -> ParserInfo InitOpts
+modifyFooter initOpts = initOpts {infoFooter = stringChunk $ T.unpack artFooter}
+
+artHeader :: Text
+artHeader = [text|
+$endLine
+                                                  ___
+                                                /  .  \
+                                               │\_/│   │
+                                               │   │  /│
+  __________________________________________________-' │
+ ╱                                                     │
+╱   .-.                                                │
+│  /   \                                               │
+│ |\_.  │ hs-init — tool for creating Haskell projects │
```

Comment:
Need to replace `hs-init` here with `summoner`... Unfortunately `summoner` is one letter longer... :disappointed: